### DISCUSSION TITLE
Can't use fetch() when using SSR

### DISCUSSION THREAD

**Discussion by Oreeeee** [sveltejs/kit#12064] [Q&A] [answered]
I'm using SvelteKit to render static HTML that is served by my custom backend (Litestar) and I want to fetch some data from it on load. However, when I'm using fetch() in `script` tag I get an error when build the static site
```
3.289 TypeError [Error]: Failed to parse URL from /data/started-time
3.289     at Object.fetch (node:internal/deps/undici/undici:16289:11)
3.289 Emitted 'error' event on Worker instance at:
3.289     at [kOnErrorMessage] (node:internal/worker:314:10)
3.289     at [kOnMessage] (node:internal/worker:325:37)
3.289     at MessagePort.<anonymous> (node:internal/worker:225:57)
3.289     at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
3.289     at exports.emitMessage (node:internal/per_context/messageport:23:28) {
3.289   [cause]: TypeError [ERR_INVALID_URL]: Invalid URL
3.289       at new NodeError (node:internal/errors:405:5)
3.289       at new URL (node:internal/url:676:13)
3.289       at new Request (node:internal/deps/undici/undici:7129:25)
3.289       at fetch2 (node:internal/deps/undici/undici:15371:25)
3.289       at Object.fetch (node:internal/deps/undici/undici:16287:18)
3.289       at fetch (node:internal/process/pre_execution:281:25)
3.289       at file:///app/www/.svelte-kit/output/server/entries/pages/_layout.svelte.js:5:3
3.289       at Object.$$render (file:///app/www/.svelte-kit/output/server/chunks/ssr.js:84:18)
3.289       at file:///app/www/.svelte-kit/output/server/entries/pages/_layout.svelte.js:16:56
3.289       at Object.$$render (file:///app/www/.svelte-kit/output/server/chunks/ssr.js:84:18) {
3.289     input: '/data/started-time',
3.289     code: 'ERR_INVALID_URL'
3.289   }
3.289 }
3.289 
3.289 Node.js v18.19.0
```

I can workaround this issue by putting it inside something, like setInterval, but it doesn't work for stuff that I want to fetch once, on page load that is variable on the server side. Can someone help?

**Comment by teemingc** [comment, marked as answer]
> 3.289   [cause]: TypeError [ERR_INVALID_URL]: Invalid URL

I'm assuming the error is due to the use of the Node.js `fetch` (server-side fetch) with a relative URL ("/data/started-time"?). Only browsers can make relative URL fetch requests. On the server, it must be an absolute URL (e.g., http://localhost:8000/data/started-time).

Instead, you should fetch the data inside a SvelteKit `load` function. Read more:
* https://kit.svelte.dev/docs/load
* https://kit.svelte.dev/docs/load#making-fetch-requests

**Reply by Oreeeee** [reply]
Yeah, I was using SvelteKit as a static site generator and that site was served by my backend. Now I switched to using separate frontend and backend and now I'm using Node to serve SvelteKit. I got it fixed by loading the data using `load` and later refreshing it in `setInterval` on the client.
